components/browser/session/README.md

```diff
@@ -12,6 +12,107 @@ Use gradle to download the library from JCenter:
 implementation "org.mozilla.components:session:{latest-version}
 ```
 
+### SessionManager
+
+`SessionManager` contains the core functionality of this component. It provides functionality to hold all active browser sessions and link them to their corresponding engine sessions. A `SessionManager` can be created by providing an [Engine](../../concept/engine/README.md) instance:
+
+```Kotlin
+val sessionManager = SessionManager(engine)
+
+// Sessions can be added and removed
+val session = Session("https://mozilla.org")
+sessionManager.add(session)
+sessionManager.remove(session)
+sessionManager.removeAll()
+
+// Sessions can be selected (i.e. for switching tabs)
+sessionManager.select(session)
+
+// Sessions can be retrieved in various ways
+val selectedSession = sessionManager.selectedSession
+val sessions = sessionManager.sessions
+val engineSession = sessionManager.getEngineSession(session) 
+val selectedEngineSession = sessionManager.getEngineSession()
+
+// Engine sessions can automatically be created if none exists for the provided session
+val engineSession = sessionManager.getOrCreateEngineSession(session)
+val selectedEngineSession = sessionManager.getOrCreateEngineSession()
+
+// Changes to a SessionManager can be observed
+sessionManager.register(object : SessionManager.Observer {
+    onSessionSelected(session: Session) {
+        // The provided session was selected. 
+    }
+    onSessionAdded(session: Session) {
+    	// The provided session was added.
+    }
+    onSessionRemoved(session: Session) {
+    	// The provided session was removed.
+    }
+    onAllSessionsRemoved() {
+    	// The session manager has been cleared (removeAll has been called).
+    }
+})
+```
+
+### Session
```

Comment:
Small nit: I'd move the Session section first since this explains what it is, and then have the SessionManager follow to show how sessions are managed.

Reply:
Good point. I will update this.